### PR TITLE
Added Automation Account diagnostic log rules

### LINKED ISSUE #1075: Automation accounts should enable diagnostic logs
# Rule request

Automation accounts should enable the following diagnostic logs:

- JobLogs
- JobStreams
- DSCNodeStatus
- Metrics

## Applies to the following

The rule applies to the following:

- Resource type: **Microsoft.Automation/automationAccounts**

## Additional context

[Template reference](https://docs.microsoft.com/en-us/azure/templates/microsoft.automation/automationaccounts?tabs=bicep)

### PR DESCRIPTION
**Pull request by ArmaanMcleod**
## PR Summary

Fix #1075

Added `Azure.Automation.AuditLogs` and `Azure.Automation.PlatformLogs` rules. 

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by ArmaanMcleod** [comment]
@BernieWhite I'm facing a weird issue where these rules seem to not pick up the diagnostic rules I have in https://github.com/ArmaanMcleod/PSRule.Rules.Azure/blob/automation-diagnostic-logs-rules/tests/PSRule.Rules.Azure.Tests/Resources.Automation.json.

So all of them are reporting as `$LocalizedData.DiagnosticSettingsNotConfigured`. 

If I run `GetSubResources` locally like so it seems to work:

```powershell
$TargetObject = (Get-Content -Path .\tests\PSRule.Rules.Azure.Tests\Resources.Automation.json -Raw | ConvertFrom-Json)
$diagnosticLogs = GetSubResources -ResourceType 'Microsoft.Insights/diagnosticSettings', 'Microsoft.Automation/automationAccounts/providers/diagnosticSettings'
$diagnosticLogs.Length # this gives 6
```

Do you know what the issue could be?

File: docs/en/rules/Azure.Automation.PlatformLogs.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
+---
+severity: Important
+pillar: Security
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Automation.PlatformLogs.md:3]: Maybe this should be a different pillar?

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod Looks like your worked out the issue. One question otherwise all good.

File: docs/en/rules/Azure.Automation.PlatformLogs.md (hunk 1 above)
Comment (ArmaanMcleod) [reply docs/en/rules/Azure.Automation.PlatformLogs.md:3]: nice spot, will change this. Thanks 🙂

**Comment by ArmaanMcleod** [comment]
> @ArmaanMcleod Looks like your worked out the issue. One question otherwise all good.

All good @BernieWhite. Had a couple issues with the rules but resolved them. Also changed the pillar for `Azure.Automation.PlatformLogs` to Operational Excellence, which matches up with the other rules 🙂

**Comment by BernieWhite** [comment]
> All good @BernieWhite. Had a couple issues with the rules but resolved them. Also changed the pillar for `Azure.Automation.PlatformLogs` to Operational Excellence, which matches up with the other rules 🙂

If there is something that we need to take back to improve PSRule then lets log it.

**Pull request merged**